<launch>
  <arg name="model" default="$(find xacro)/xacro '$(find car_model)/urdf/zhongqi.xacro'" />
  <param name="robot_description" command="$(arg model) roboname:='smart'" />
   
  <!--node name="zhongqi_driver" pkg="zhongqi_driver" type="zhongqi_driver_node" output="screen">
    <param name="port_name" value="/dev/ucar" />
    <param name="control_rate" value="10" />
    <param name="check_pos_rate" value="1" />
    <param name="baud_rate" value="115200" />
    <param name="wheelbase" value="1.07" />
    <param name="tread" value="0.685" />
    <param name="wheel_diameter" value="0.37" />
    <param name="maxsteer" value="1" />
    <param name="steer_radius" value="true" />
    <param name="publish_tf" value="true" />
    <param name="check" value="false" />
    <param name="odom_frame" value="wheel_odom"/>
    <param name="base_frame" value="base_footprint"/>
    <param name="vth_c" value="1.0" />
    <param name="vx_c" value="1.146" />
    <param name="vy_c" value="1" />
    <param name="vth_c" value="1" />
  </node-->

  
  <node name="cmd_vel_to_ackermann" pkg="zhongqi_driver" type="cmd_vel_to_ackermann_drive.py" output="screen">
  </node>

  <!--node name="keyboard_control" pkg="keyboard_control" type="keyboard_control_node" output="screen">
    <param name="linear_init" value="0" />
    <param name="linear_max" value="1" />
    <param name="linear_step" value="0.05" />

    <param name="angular_init" value="0.3" />
    <param name="angular_max" value="0.3" />
    <param name="angular_step" value="0.01" />
  </node-->
  
  <!--node name="location_fusion" pkg="location_fusion" type="simple_fusion">
      <param name="map_frame" value="map"/>
      <param name="odom_frame" value="odom"/>
      <param name="base_link_frame" value="base_link"/>
      <param name="tag_frame" value="tag"/>

      <param name="buffer_size" value="50"/>
      <param name="distance_interval" value="0.1"/>
      <param name="rate" value="200"/>
  </node-->
    
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

  <!--node name="rviz" pkg="rviz" type="rviz" args="-d $(find zhongqi_driver)/rviz/smart.rviz" /-->
</launch>
